core/src/main/java/org/apache/accumulo/core/client/rfile/RFileScanner.java

```diff
@@ -364,17 +366,20 @@ public SamplerConfiguration getSamplerConfiguration() {
       if (opts.useSystemIterators) {
         SortedSet<Column> cols = this.getFetchedColumns();
         families = LocalityGroupUtil.families(cols);
-        iterator = IteratorUtil.setupSystemScanIterators(iterator, cols, getAuthorizations(),
+        iterator = SystemIteratorUtil.setupSystemScanIterators(iterator, cols, getAuthorizations(),
             EMPTY_BYTES, tableConf);
       }
 
       try {
         if (opts.tableConfig != null && opts.tableConfig.size() > 0) {
-          iterator = IteratorUtil.loadIterators(IteratorScope.scan, iterator, null, tableConf,
-              serverSideIteratorList, serverSideIteratorOptions, new IterEnv());
+          IterLoad il = IterConfigUtil.loadIterConf(IteratorScope.scan, serverSideIteratorList,
```

Comment:
The unit test for this is failing.   Seems like an unmodifiable list is passed in and it tries to modifiy it.